elastic/beats
# [Filebeat] Rename awscloudwatch input to aws-cloudwatch input

## What does this PR do?

This PR is to rename `awscloudwatch` input to `aws-cloudwatch` to follow the naming convention for Filebeat inputs.
After this PR get cherry-picked into 7.x, we can remove support of `awscloudwatch`. Or create a separate issue to remove old names in 8.0.

Deprecation warning log message is also added to this PR:
```
2020-10-28T12:41:21.556-0600    WARN    [aws-cloudwatch]        awscloudwatch/input.go:93       awscloudwatch input name is deprecated, please use aws-cloudwatch instead
```

## Checklist




## How to test this PR locally

Make sure both `awscloudwatch` and `aws-cloudwatch` input work:
```
filebeat.inputs:
  - type: awscloudwatch
    credential_profile_name: elastic-beats
    log_group_arn: arn:aws:logs:us-east-1:428152502467:log-group:test:*
    region: us-east-1
    scan_frequency: 30s
    start_position: beginning
    api_timeout: 5m
```
and 
```
filebeat.inputs:
  - type: aws-cloudwatch
    credential_profile_name: elastic-beats
    log_group_arn: arn:aws:logs:us-east-1:428152502467:log-group:test:*
    region: us-east-1
    scan_frequency: 30s
    start_position: beginning
    api_timeout: 5m
```

## Related issues

- Relates https://github.com/elastic/beats/pull/22214
